src/pcre2_jit_neon_inc.h

```diff
@@ -82,7 +82,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #  endif
 # endif
 
-static sljit_u8* SLJIT_FUNC FF_FUN(sljit_u8 *str_end, sljit_u8 *str_ptr, sljit_uw offs1, sljit_uw offs2, sljit_uw chars)
+static sljit_u8* SLJIT_FUNC FF_FUN(sljit_u8 *str_end, sljit_u8 **str_ptr, sljit_uw offs1, sljit_uw offs2, sljit_uw chars)
```

Comment:
This function should return with str_ptr.

Reply:
it does, but only if the match succeeds.

the str_ptr "reference" is needed to keep track of all the progress made if there is no match, specially as it loops over until the end_ptr looking for matches.

Reply:
Something is not right here. This variant (with two offsets) of the code should not be used when partial matching is enabled.

Reply:
Not sure if I follow, but there s plenty wrong with the arm64 simd helper; this ONLY tackles the issues that were reported and that affect mostly the single character + mask, one offset function.

 Since they all use the same template function, the changes to the second parameter to be a pointe to pointer so it can be used as IO propagate to all functions though, which also include the one with 4 characters and 2 offsets which wasn't being triggered AFAIK but that would be otherwise unaffected by the change (except that it might perform slightly better, just like the equivalent one from other architectures).